.github/actions/build-infinispan/action.yml

```diff
@@ -0,0 +1,37 @@
+name: Build Infinispan
+description: Builds Infinispan and Maven repository with all required artifacts
+
+runs:
+  using: composite
+  steps:
+    - name: Setup Java
+      uses: ./.github/actions/setup-java
+
+    - name: Build Infinispan
+      shell: bash
+      run: |
+        # Ensure this plugin is built first to avoid warnings in the build
+        ./mvnw install -Pdistribution -s maven-settings.xml
+        # By using "dependency:resolve", it will download all dependencies used in later stages for running the tests
+        ./mvnw install dependency:resolve -V -e -DskipTests -DexcludeGroupIds=org.infinispan -Dsilent=true -s maven-settings.xml
+
+    - name: Compress Infinispan Maven artifacts
+      shell: bash
+      run: |
+        tar -C ~/ --use-compress-program zstd -cf m2-infinispan.tzts \
+        --exclude '*.tar.gz' \
+        .m2/repository/org/infinispan
+
+    - name: Upload Infinispan Maven artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: m2-infinispan.tzts
```

Comment:
is the `name` private to the workflow or is it possible the concurrent PRs running this action will overwrite the upload?

Reply:
I believe it's private to the workflow. If you look at the url of the uploaded artifacts, each one has a unique artifact id which is a sub-resource of the individual run. Keycloak uses the same name on each PR without issue.